src/quicklist.c

```diff
@@ -869,7 +869,7 @@ REDIS_STATIC void _quicklistInsert(quicklist *quicklist, quicklistEntry *entry,
     if (after && (entry->offset == node->count)) {
         D("At Tail of current ziplist");
         at_tail = 1;
-        if (!_quicklistNodeAllowInsert(node->next, fill, sz)) {
+        if (full && !_quicklistNodeAllowInsert(node->next, fill, sz)) {
```

Comment:
The question is why avoid setting `full_next` if we're full. It can't hurt. Is this just an optimization?
If `full` is true we never reach any code that checks full next anyway.

The comment above explicitly states that the design was to `/* Populate accounting flags for easier boolean checks later */` so unless we change this design I suggest avoiding this fix.

Reply:
Thanks  a lot for your suggesttions. 
Yes,  it can't hurt since `_quicklistNodeAllowInsert` is fast. I will remove it. 
And it is a good idea to use `avail_*` to replace `full_*`, which exactly make code more clearly.